pkg/transforms/http_test.go

```diff
@@ -116,9 +117,10 @@ func TestHTTPPostWithSecrets(t *testing.T) {
 
 		w.WriteHeader(http.StatusOK)
 
-		// this should be here on successful secret retrieval
-		if assert.Equal(t, r.Header.Get("Secret-Header-Name"), "value") {
-			t.Errorf("Unexpected header name received %s, expected %s", r.Header.Get("Content-type"), "application/json")
+		actualValue := r.Header.Get("Secret-Header-Name")
+		if actualValue != "" {
```

Comment:
this looks different than the original logic: the original logic does test the case of missing key; as of now this skips that case.   Is this ok? or should we add another test case for missing key?

Reply:
It was causing failure because there are test cases where the key is specifically left out. See the test cases in block below this code.

Reply:
got it. LGTM then.